### PR TITLE
DOC: rework API docs around the new functional stuff 

### PR DESCRIPTION
**Pull request by martinfleis**

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -4,101 +4,152 @@
...
 .. currentmodule:: momepy
 
-
 momepy API reference
 ======================
 
-elements
---------
+The current version of momepy includes two implementations of most of the functionality
+due to the ongoing migration period moving from the legacy class-based API to new
```
Comment (jGaboardi) [inline docs/api.rst:11]:
```suggestion
due to the ongoing migration period moving from the legacy class-based API to a new
```

File: docs/api.rst [hunk 2]
```diff
@@ -4,101 +4,152 @@
...
+   neighbor_distance
+   neighbors
+   orientation
+   shared_walls
+   street_alignment
+
+Measuring intensity
+-------------------
+
+A set of functions to measure intensity characters:
```
Comment (jGaboardi) [inline docs/api.rst:108]: Should this be "characteristics"?

**Comment by martinfleis** [comment]
Here's the outcome https://momepy--597.org.readthedocs.build/en/597/api.html

File: docs/api.rst [hunk 3]
```diff
@@ -4,101 +4,152 @@
...
-   Reached
-
-graph
-------------
+   :toctree: api/
+
+   courtyards
+   node_density
+
+Note that additional intensity characters can be directly derived using :meth:`libpysal.graph.Graph.describe`
```
Comment (jGaboardi) [inline docs/api.rst:116]: Same here.

File: docs/api.rst [hunk 4]
```diff
@@ -4,101 +4,152 @@
...
+   describe_reached_agg
+   gini
+   percentile
+   shannon
+   simpson
+   theil
+   values_range
+
+
+Underlying components of :func:`shannon` and :func:`simpson` are also exposed for a direct use:
```
Comment (jGaboardi) [inline docs/api.rst:137]:
```suggestion
Underlying components of :func:`shannon` and :func:`simpson` are also exposed for direct use:
```

File: docs/api.rst [hunk 5]
```diff
@@ -115,52 +166,44 @@ graph
...
-   consolidate_intersections
-   CheckTessellationInput
    extend_lines
-   FaceArtifacts
    remove_false_nodes
-   preprocess
+   consolidate_intersections
    roundabout_simplification
 
+Additionally, there are methods for data assesment.
```
Comment (jGaboardi) [inline docs/api.rst:194]:
```suggestion
Additionally, there are methods for data assessment.
```

File: docs/api.rst [hunk 6]
```diff
@@ -115,52 +166,44 @@ graph
...
 
-   gdf_to_nx
-   limit_range
-   nx_to_gdf
-   unique_id
+   CheckTessellationInput
+   FaceArtifacts
+
+
+Furhter analysis can be done directly using methods available in :class:`libpysal.graph.Graph`.
```
Comment (jGaboardi) [inline docs/api.rst:203]:
```suggestion
Further analysis can be done directly using methods available in :class:`libpysal.graph.Graph`.
```

File: docs/user_guide/getting_started.ipynb
Comment (jGaboardi) [inline docs/user_guide/getting_started.ipynb:1]: * **Dependencies**
  * probably can remove `descartes`, no?
* **Simple examples**
  * "... using *the* bubenec dataset"
* After cell 3
  * "Momepy uses classes for each measurable character" -- Is this still true with the refactor?

This notebook still uses the class API. Maybe I'm looking at an older version.

Comment (jGaboardi) [reply docs/user_guide/getting_started.ipynb:1]: Or are we keeping the class-based API stuff in "Getting Started" for now?

Comment (martinfleis) [reply docs/user_guide/getting_started.ipynb:1]: I'll get to the notebooks later. I just changed the links to API sections to avoid sphinx warnings.

**Comment by jGaboardi** [comment]
I gave the rendered docs a once over -- looking good.

File: docs/api.rst (hunk 2 above)
Comment (martinfleis) [reply docs/api.rst:108]: These are "morphometric characters", which is a term used in literature.

**Review by jGaboardi — APPROVED** [review]

**Pull request merged**